<!doctype html>
<html lang=en>
  <head>
    <title>KeyError: 'value'
 // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css">
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script>
      var CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "fO1h2ktLnSv4O3fYRjeC";
    </script>
  </head>
  <body style="background-color: #fff">
    <div class="debugger">
<h1>KeyError</h1>
<div class="detail">
  <p class="errormsg">KeyError: &#39;value&#39;
</p>
</div>
<h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
<div class="traceback">
  <h3></h3>
  <ul><li><div class="frame" id="frame-140105589826624">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">2548</em>,
      in <code class="function">__call__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>def __call__(self, environ: dict, start_response: t.Callable) -&gt; t.Any:</pre>
<pre class="line before"><span class="ws">        </span>&#34;&#34;&#34;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
<pre class="line before"><span class="ws">        </span>&#34;&#34;&#34;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)
<span class="ws">        </span>       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre></div>
</div>

<li><div class="frame" id="frame-140105589827200">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">2528</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)
<span class="ws">                </span>           ^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-140105589827056">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask_cors/extension.py"</cite>,
      line <em class="line">165</em>,
      in <code class="function">wrapped_function</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span># Wrap exception handlers with cross_origin</pre>
<pre class="line before"><span class="ws">        </span># These error handlers will still respect the behavior of the route</pre>
<pre class="line before"><span class="ws">        </span>if options.get(&#39;intercept_exceptions&#39;, True):</pre>
<pre class="line before"><span class="ws">            </span>def _after_request_decorator(f):</pre>
<pre class="line before"><span class="ws">                </span>def wrapped_function(*args, **kwargs):</pre>
<pre class="line current"><span class="ws">                    </span>return cors_after_request(app.make_response(f(*args, **kwargs)))
<span class="ws">                    </span>                                            ^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">                </span>return wrapped_function</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">            </span>if hasattr(app, &#39;handle_exception&#39;):</pre>
<pre class="line after"><span class="ws">                </span>app.handle_exception = _after_request_decorator(</pre>
<pre class="line after"><span class="ws">                    </span>app.handle_exception)</pre></div>
</div>

<li><div class="frame" id="frame-140105589826912">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">2525</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error: t.Optional[BaseException] = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()
<span class="ws">                </span>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-140105589826768">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">1822</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)
<span class="ws">            </span>     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
<pre class="line after"><span class="ws">        </span>self,</pre>
<pre class="line after"><span class="ws">        </span>rv: t.Union[ft.ResponseReturnValue, HTTPException],</pre></div>
</div>

<li><div class="frame" id="frame-140105589827344">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask_cors/extension.py"</cite>,
      line <em class="line">165</em>,
      in <code class="function">wrapped_function</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span># Wrap exception handlers with cross_origin</pre>
<pre class="line before"><span class="ws">        </span># These error handlers will still respect the behavior of the route</pre>
<pre class="line before"><span class="ws">        </span>if options.get(&#39;intercept_exceptions&#39;, True):</pre>
<pre class="line before"><span class="ws">            </span>def _after_request_decorator(f):</pre>
<pre class="line before"><span class="ws">                </span>def wrapped_function(*args, **kwargs):</pre>
<pre class="line current"><span class="ws">                    </span>return cors_after_request(app.make_response(f(*args, **kwargs)))
<span class="ws">                    </span>                                            ^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">                </span>return wrapped_function</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">            </span>if hasattr(app, &#39;handle_exception&#39;):</pre>
<pre class="line after"><span class="ws">                </span>app.handle_exception = _after_request_decorator(</pre>
<pre class="line after"><span class="ws">                    </span>app.handle_exception)</pre></div>
</div>

<li><div class="frame" id="frame-140105589827488">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">1820</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()
<span class="ws">                </span>     ^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre></div>
</div>

<li><div class="frame" id="frame-140105589827632">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask/app.py"</cite>,
      line <em class="line">1796</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>and req.method == &#34;OPTIONS&#34;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line before"><span class="ws">        </span>view_args: t.Dict[str, t.Any] = req.view_args  # type: ignore[assignment]</pre>
<pre class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
<span class="ws">        </span>       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
<pre class="line after"><span class="ws">        </span>&#34;&#34;&#34;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-140105589827776">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/flask_httpauth.py"</cite>,
      line <em class="line">172</em>,
      in <code class="function">decorated</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                        </span>except TypeError:</pre>
<pre class="line before"><span class="ws">                            </span>return self.auth_error_callback()</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">                    </span>g.flask_httpauth_user = user if user is not True \</pre>
<pre class="line before"><span class="ws">                        </span>else auth.username if auth else None</pre>
<pre class="line current"><span class="ws">                </span>return self.ensure_sync(f)(*args, **kwargs)
<span class="ws">                </span>       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">            </span>return decorated</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>if f:</pre>
<pre class="line after"><span class="ws">            </span>return login_required_internal(f)</pre>
<pre class="line after"><span class="ws">        </span>return login_required_internal</pre></div>
</div>

<li><div class="frame" id="frame-140105589827920">
  <h4>File <cite class="filename">"/app/endpoints.py"</cite>,
      line <em class="line">227</em>,
      in <code class="function">get_grouptask</code></h4>
  <div class="source "><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>print(f&#34;{len(results)} of { len(group_result.results) } tasks ready.&#34;)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span>if results:</pre>
<pre class="line before"><span class="ws">        </span># first summarize the results into 1 geojson</pre>
<pre class="line current"><span class="ws">        </span>results = summarize_multiple_geojsons_to_one([result[&#34;geojson&#34;] for result in results])
<span class="ws">        </span>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>if result_format == &#34;png&#34;:</pre>
<pre class="line after"><span class="ws">            </span>print(&#34;converting result to png&#34;)</pre>
<pre class="line after"><span class="ws">            </span>results = convert_result_to_png(results)</pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>

<li><div class="frame" id="frame-140105589828064">
  <h4>File <cite class="filename">"/app/wind/data.py"</cite>,
      line <em class="line">280</em>,
      in <code class="function">summarize_multiple_geojsons_to_one</code></h4>
  <div class="source "><pre class="line before"><span class="ws">        </span>crs=&#34;EPSG:4326&#34;,</pre>
<pre class="line before"><span class="ws">        </span>columns=list(all_features[0][&#34;properties&#34;].keys()) + [&#34;geometry&#34;]</pre>
<pre class="line before"><span class="ws">    </span>)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span># dissolve polygons with the same value in the &#34;value&#34; column</pre>
<pre class="line current"><span class="ws">    </span>results_gdf.dissolve(by=&#39;value&#39;)
<span class="ws">    </span>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>return json.loads(results_gdf.to_json())</pre></div>
</div>

<li><div class="frame" id="frame-140105589828208">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/geopandas/geodataframe.py"</cite>,
      line <em class="line">1676</em>,
      in <code class="function">dissolve</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>if not dropna:  # If they passed a non-default dropna value</pre>
<pre class="line before"><span class="ws">                </span>warnings.warn(&#34;dropna kwarg is not supported for pandas &lt; 1.1.0&#34;)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span># Process non-spatial component</pre>
<pre class="line before"><span class="ws">        </span>data = self.drop(labels=self.geometry.name, axis=1)</pre>
<pre class="line current"><span class="ws">        </span>aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc)
<span class="ws">        </span>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">        </span>aggregated_data.columns = aggregated_data.columns.to_flat_index()</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span># Process spatial component</pre>
<pre class="line after"><span class="ws">        </span>def merge_geometries(block):</pre>
<pre class="line after"><span class="ws">            </span>merged_geom = block.unary_union</pre></div>
</div>

<li><div class="frame" id="frame-140105589828352">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/pandas/core/frame.py"</cite>,
      line <em class="line">8402</em>,
      in <code class="function">groupby</code></h4>
  <div class="source library"><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if level is None and by is None:</pre>
<pre class="line before"><span class="ws">            </span>raise TypeError(&#34;You have to supply one of &#39;by&#39; and &#39;level&#39;&#34;)</pre>
<pre class="line before"><span class="ws">        </span>axis = self._get_axis_number(axis)</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">        </span>return DataFrameGroupBy(
<span class="ws">        </span>       </pre>
<pre class="line after"><span class="ws">            </span>obj=self,</pre>
<pre class="line after"><span class="ws">            </span>keys=by,</pre>
<pre class="line after"><span class="ws">            </span>axis=axis,</pre>
<pre class="line after"><span class="ws">            </span>level=level,</pre>
<pre class="line after"><span class="ws">            </span>as_index=as_index,</pre></div>
</div>

<li><div class="frame" id="frame-140105589828496">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/pandas/core/groupby/groupby.py"</cite>,
      line <em class="line">965</em>,
      in <code class="function">__init__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>self.dropna = dropna</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>if grouper is None:</pre>
<pre class="line before"><span class="ws">            </span>from pandas.core.groupby.grouper import get_grouper</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">            </span>grouper, exclusions, obj = get_grouper(
<span class="ws">            </span>                           </pre>
<pre class="line after"><span class="ws">                </span>obj,</pre>
<pre class="line after"><span class="ws">                </span>keys,</pre>
<pre class="line after"><span class="ws">                </span>axis=axis,</pre>
<pre class="line after"><span class="ws">                </span>level=level,</pre>
<pre class="line after"><span class="ws">                </span>sort=sort,</pre></div>
</div>

<li><div class="frame" id="frame-140105589828640">
  <h4>File <cite class="filename">"/usr/local/lib/python3.11/site-packages/pandas/core/groupby/grouper.py"</cite>,
      line <em class="line">888</em>,
      in <code class="function">get_grouper</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                    </span>raise ValueError(f&#34;Grouper for &#39;{name}&#39; not 1-dimensional&#34;)</pre>
<pre class="line before"><span class="ws">                </span>exclusions.add(name)</pre>
<pre class="line before"><span class="ws">            </span>elif obj._is_level_reference(gpr, axis=axis):</pre>
<pre class="line before"><span class="ws">                </span>in_axis, level, gpr = False, gpr, None</pre>
<pre class="line before"><span class="ws">            </span>else:</pre>
<pre class="line current"><span class="ws">                </span>raise KeyError(gpr)
<span class="ws">                </span>^^^^^^^^^^^^^^^^^^^</pre>
<pre class="line after"><span class="ws">        </span>elif isinstance(gpr, Grouper) and gpr.key is not None:</pre>
<pre class="line after"><span class="ws">            </span># Add key to exclusions</pre>
<pre class="line after"><span class="ws">            </span>exclusions.add(gpr.key)</pre>
<pre class="line after"><span class="ws">            </span>in_axis = False</pre>
<pre class="line after"><span class="ws">        </span>else:</pre></div>
</div>
</ul>
  <blockquote>KeyError: &#39;value&#39;
</blockquote>
</div>

<div class="plain">
    <p>
      This is the Copy/Paste friendly version of the traceback.
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 2548, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 2528, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask_cors/extension.py&#34;, line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 2525, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask_cors/extension.py&#34;, line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 1820, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask/app.py&#34;, line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/flask_httpauth.py&#34;, line 172, in decorated
    return self.ensure_sync(f)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/app/endpoints.py&#34;, line 227, in get_grouptask
    results = summarize_multiple_geojsons_to_one([result[&#34;geojson&#34;] for result in results])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/app/wind/data.py&#34;, line 280, in summarize_multiple_geojsons_to_one
    results_gdf.dissolve(by=&#39;value&#39;)
  File &#34;/usr/local/lib/python3.11/site-packages/geopandas/geodataframe.py&#34;, line 1676, in dissolve
    aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/pandas/core/frame.py&#34;, line 8402, in groupby
    return DataFrameGroupBy(
           ^^^^^^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/pandas/core/groupby/groupby.py&#34;, line 965, in __init__
    grouper, exclusions, obj = get_grouper(
                               ^^^^^^^^^^^^
  File &#34;/usr/local/lib/python3.11/site-packages/pandas/core/groupby/grouper.py&#34;, line 888, in get_grouper
    raise KeyError(gpr)
KeyError: &#39;value&#39;
</textarea>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2548, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
                                                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_httpauth.py", line 172, in decorated
    return self.ensure_sync(f)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/endpoints.py", line 227, in get_grouptask
    results = summarize_multiple_geojsons_to_one([result["geojson"] for result in results])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/wind/data.py", line 280, in summarize_multiple_geojsons_to_one
    results_gdf.dissolve(by='value')
  File "/usr/local/lib/python3.11/site-packages/geopandas/geodataframe.py", line 1676, in dissolve
    aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pandas/core/frame.py", line 8402, in groupby
    return DataFrameGroupBy(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pandas/core/groupby/groupby.py", line 965, in __init__
    grouper, exclusions, obj = get_grouper(
                               ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pandas/core/groupby/grouper.py", line 888, in get_grouper
    raise KeyError(gpr)
KeyError: 'value'


-->
